modules/arch-kubernetes-infrastructure.adoc

```diff
@@ -0,0 +1,305 @@
+static pods. RPM-based etcd is still supported on etcd hosts that are not also
+masters.
+
+In addition, the node components `openshift-sdn` and
+`openvswitch` are now run using a DaemonSet instead of a `systemd` service.
+
+.Control plane host architecture changes
+image::ocp310-archupgrade.png["Control plane host architecture changes"]
+
+ifdef::openshift-enterprise,openshift-origin[]
+[discrete]
+[id='control-plane-static-pods-mirror-pods-{context}']
+==== Mirror Pods
+
+The kubelet on master nodes automatically creates _mirror pods_ on the API
+server for each of the control plane static pods so that they are visible in the
+cluster in the `kube-system` project. Manifests for these static pods are
+installed by default by the `openshift-ansible` installer, located in the
+`/etc/origin/node/pods` directory on the master host.
+
+These pods have the following `hostPath` volumes defined:
+
+[horizontal]
+`/etc/origin/master`:: Contains all certificates, configuration files, and the `admin.kubeconfig` file.
+`/var/lib/origin`:: Contains volumes and potential core dumps of the binary.
+`/etc/origin/cloudprovider`:: Contains cloud provider specific configuration (AWS, Azure, etc.).
+`/usr/libexec/kubernetes/kubelet-plugins`:: Contains additional third party volume plug-ins.
+`/etc/origin/kubelet-plugins`:: Contains additional third party volume plug-ins for system containers.
+
+The set of operations you can do on the static pods is limited.
+endif::[]
+
+ifdef::openshift-origin,openshift-enterprise,openshift-dedicated[]
+[id='high-availability-masters-{context}']
+
+=== High Availability Masters
+
+The availability of running applications remains if the master or any of its services fail.
+However, failure of master services reduces the ability of the system to respond to
+application failures or creation of new applications.
+endif::[]
+ifdef::openshift-origin,openshift-enterprise[]
+You can optionally configure your masters for high
+availability (HA) to ensure that the cluster has no single point of failure.
+
+To mitigate concerns about availability of the master, two activities are
+recommended:
+
+1. A https://en.wikipedia.org/wiki/Runbook[runbook] entry should be created for
+reconstructing the master. A runbook entry is a necessary backstop for any
+highly-available service. Additional solutions merely control the frequency
+that the runbook must be consulted. For example, a cold standby of the master
+host can adequately fulfill SLAs that require no more than minutes of downtime
+for creation of new applications or recovery of failed application components.
+
+2. Use a high availability solution to configure your masters and ensure that the
+cluster has no single point of failure. The cluster
+installation documentation provides specific examples using the `native` HA method and
+configuring HAProxy. You can also take the concepts and apply them towards your
+existing HA solutions using the `native` method instead of HAProxy.
+
+[NOTE]
```

Comment:
@ariordan-redhat, this note was added to the 3.x stream on https://github.com/openshift/openshift-docs/pull/13783. Will you confirm that it's valid in 4.0 for me?